### DISCUSSION TITLE
Got an error while trying to run for the first time

### DISCUSSION THREAD

**Discussion by skydiver** [jorge-menjivar/unSAGED#11] [Q&A]
I'm getting this error:

![Screenshot 2023-07-04 at 00 41 04](https://github.com/jorge-menjivar/unSAGED/assets/578784/b3a4d334-e12c-49cc-bb99-e7bf919632d8)

I'm using Azure ... my .env is following:

```
NEXT_PUBLIC_SUPABASE_URL=****
NEXT_PUBLIC_SUPABASE_ANON_KEY=****

OPENAI_API_KEY=****
OPENAI_API_URL=****
OPENAI_API_TYPE=azure
OPENAI_API_VERSION=2023-03-15-preview
DEFAULT_MODEL=gpt-35-turbo
```

any ideas?

Thanks!

**Comment by skydiver** [comment]
**Update:**

Problem seems to be there is no defined model for Azure in `types/ai-models.ts`

First, model name is `gpt-35-turbo` and second deployment name needs to be defined in .env (because it's a dynamic value)

My final solution was to add new row to `PossibleAiModels`:

```
  'gpt-35-turbo': {
    id: String(process.env.NEXT_PUBLIC_DEPLOYMENT_ID),
    maxLength: 12000,
    tokenLimit: 4000,
    requestLimit: 3000,
    vendor: 'OpenAI',
  },
```

**Reply by jorge-menjivar** [reply]
Do you mind creating a pull request with your fix? Thanks!

**Reply by skydiver** [reply]
done https://github.com/jorge-menjivar/unSAGED/pull/12
